### PR TITLE
Gdr 2419

### PR DESCRIPTION
**Pull request by gladkia**
# Description
Improve pkgdown site
## What changed?
- updated news and added reference sections
- Related JIRA issue:  GDR-2419
- pkgdown site link: https://gdrplatform.github.io/gDRtestData/

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -2,3 +2,16 @@ url: https://gdrplatform.github.io/gDRtestData/
 template:
   bootstrap: 5
 
+reference:
+- title: Generate data
```
Comment (j-smola) [inline _pkgdown.yml:6]: Could you add short description of these function groups?

**Review by j-smola — COMMENTED** [review]
Proposal - maybe instead of grouping by file we could proposed group by functionality (see also our vignette):
1. Generate synthetic data: create_synthetic_cell_lines(), create_synthetic_drugs(), add_data_replicates(), add_concentration(), prepareData()
3. Generate basic synthetic result: generate_hill_coef(), generate_ec50(), generate_e_inf(), generate_response_data(), add_day0_data()
4. Generate test data for different experiment type: prepareMergedData(), prepareCodilutionData(), prepareComboMergedData(),

**Review by marcin-kam — APPROVED** [review]

**Comment by gladkia** [comment]
> Proposal - maybe instead of grouping by file we could proposed group by functionality (see also our vignette):
> 
> 1. Generate synthetic data: create_synthetic_cell_lines(), create_synthetic_drugs(), add_data_replicates(), add_concentration(), prepareData()
> 2. Generate basic synthetic result: generate_hill_coef(), generate_ec50(), generate_e_inf(), generate_response_data(), add_day0_data()
> 3. Generate test data for different experiment type: prepareMergedData(), prepareCodilutionData(), prepareComboMergedData(),

Great idea. done.

File: _pkgdown.yml (hunk 1 above)
Comment (gladkia) [reply _pkgdown.yml:6]: I think after the latest changes: 
 - titles are telling and self-explanatory
 - users can get detailed info about all three categories by looking into the 'Get started' section.

**Pull request merged**